### PR TITLE
nix repl: add :edit command

### PR DESCRIPTION
**Pull request by zimbatm**
This allows to have a repl-centric workflow to working on nixpkgs.

Usage:

    :edit <package> - heuristic that find the package file path

    :edit <path> - just open the editor on the file path

Once invoked, `nix repl` will open $EDITOR on that file path. Once the
editor exits, `nix repl` will automatically reload itself.

### CODE REVIEW

**Comment by zimbatm** [comment]
This PR borrows some code from `nix edit` which I would like to factor out but don't know where to place it.

Currently this is a feature preview. The doc also needs to be amended with the new command.

**Comment by bhipple** [comment]
This looks nice!  It'd be good to have some test cases though.  Something basic like this:

```
[nix-shell:~/src/nixpkgs]$ echo ":edit boost" | EDITOR=ls ../nix/inst/bin/nix repl .
Welcome to Nix version 2.4. Type :? for help.

Loading '.'...
Added 11053 variables.

/home/bhipple/src/nixpkgs/pkgs/development/libraries/boost/generic.nix
Loading '.'...
Added 11053 variables.
```
That can assert that it uses `$EDITOR`, finds the package, reloads `nix repl`, etc.

File: src/nix/edit.cc [hunk 1]
```diff
@@ -36,45 +36,17 @@ struct CmdEdit : InstallableCommand
...
 
         stopProgressBar();
 
+        auto args = editorFor(pos);
+
         execvp(args.front().c_str(), stringsToCharPtrs(args).data());
 
-        throw SysError("cannot run editor '%s'", editor);
+        std::string command;
+        for (const auto &arg : args) command += " '" + arg + "'";
```
Comment (bhipple) [inline src/nix/edit.cc]: Note a huge deal, but it'd be more idiomatic (and performant) to use a `std::stringstream` here.

File: src/nix/repl.cc [hunk 2]
```diff
@@ -440,6 +441,7 @@ bool NixRepl::processLine(string line)
              << "  <x> = <expr>  Bind expression to variable\n"
              << "  :a <expr>     Add attributes from resulting set to scope\n"
              << "  :b <expr>     Build derivation\n"
+             << "  :e <expr>     Open the derivation in $EDITOR\n"
```
Comment (bhipple) [inline src/nix/repl.cc]: Unrelated to this PR, but it probably makes sense to curate an (unreleased) `doc/manual/release-notes/rl-2.4.xml` that gets updated with upcoming changes as we add them.  That way @edolstra doesn't have to go through the git log and write them all himself when he wants to do a release, and users can see what unreleased changes have accumulated on master.

As an extra aside, it looks like we don't have man pages or manual documentation for the nix 2.0 CLI, so I guess this is the only place it needs to be documented?

**Pull request merged**